facades/PC/src/main/java/org/terasology/engine/Terasology.java

```diff
@@ -153,16 +154,16 @@ public static void main(String[] args) {
                 engine.run(new StateHeadlessSetup());
             } else if (loadLastGame) {
                 engine.initialize(); //initialize the managers first
-                engine.getFromEngineContext(ThreadManager.class).submitTask("loadGame", () -> {
```

Comment:
It looks like with `ThreadManager` we could give the tasks a name such that they are identifiable later, e.g., in logs. Is there a similar concept in RxJava?

Reply:
It looks like there is a subscriber [Context](https://projectreactor.io/docs/core/release/reference/#context) that is useful for things like this.

Reply:
#4839 added a `GameScheduler.scheduleParallel` method that takes a name which it uses in the same way as ThreadManager.submitTask did.

It doesn't address future use cases where we take advantage of Flux's model for pipelines, but it suffices for one-shot Runnables like this.

Reply:
should be fine for our use cases. i'm not too happy but it should be good for our use cases at this point.